mac.c

```diff
@@ -8,11 +8,11 @@
 #include <stdio.h>
 #include <stdbool.h>
 
-bool running = true;
-int ip = 0;
-int sp = -1;
+bool _running = true;
+int ip = 0; // counter for "source code"
```

Comment:
As for these I think it would be better to explain it as a set of an instructions, ip stands for instruction pointer, i.e. this number points at the instruction we are executing or in other words the index in the program array. Same goes for the stack pointer.